test/calibration/test_calibrations.py

```diff
@@ -0,0 +1,513 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Class to test the calibrations."""
+
+from datetime import datetime
+from qiskit.circuit import Parameter
+from qiskit.pulse import Drag, DriveChannel, ControlChannel, Gaussian, GaussianSquare
+from qiskit.pulse.transforms import inline_subroutines
+import qiskit.pulse as pulse
+from qiskit.test import QiskitTestCase
+from qiskit_experiments.calibration.calibrations_manager import CalibrationsManager, ParameterKey
+from qiskit_experiments.calibration.parameter_value import ParameterValue
+from qiskit_experiments.calibration.exceptions import CalibrationError
+
+
+class TestCalibrationsBasic(QiskitTestCase):
+    """Class to test the management of schedules and parameters for calibrations."""
+
+    def setUp(self):
+        """Setup a test environment."""
+        self.cals = CalibrationsManager()
+
+        self.sigma = Parameter("σ")
+        self.amp_xp = Parameter("amp")
+        self.amp_x90p = Parameter("amp")
+        self.amp_y90p = Parameter("amp")
+        self.beta = Parameter("β")
+        self.drive = DriveChannel(Parameter("ch0"))
+
+        # Define and add template schedules.
+        with pulse.build(name="xp") as xp:
+            pulse.play(Drag(160, self.amp_xp, self.sigma, self.beta), self.drive)
+
+        with pulse.build(name="xm") as xm:
+            pulse.play(Drag(160, -self.amp_xp, self.sigma, self.beta), self.drive)
+
+        with pulse.build(name="x90p") as x90p:
+            pulse.play(Drag(160, self.amp_x90p, self.sigma, self.beta), self.drive)
+
+        with pulse.build(name="y90p") as y90p:
+            pulse.play(Drag(160, self.amp_y90p, self.sigma, self.beta), self.drive)
+
+        for sched in [xp, x90p, y90p, xm]:
+            self.cals.add_schedule(sched)
```

Comment:
All the tests use `Schedule`s right now. It would be nice to have a test using `ScheduleBlock`.

Reply:
See here: d99c429